### ISSUE TITLE
A local variable named '__value' is already defined in this scope

### ISSUE THREAD

**Issue by jamesford42** [mono/CppSharp#437]
When using the macro CS_VALUE_TYPE on a type which has more than one array field, CppSharp generates code which does not build. Here is the code.

C++:
```C++
struct CS_VALUE_TYPE UnpackedPacket
{
	char fromOnlineId[SCE_NP_ONLINEID_MAX_LENGTH];
	char toOnlineId[SCE_NP_ONLINEID_MAX_LENGTH];	
	uint8_t buffer[kPacketPayloadSize];
	size_t size;
};
```

C# (problem method):
```C#
public UnpackedPacket(global::System.IntPtr native, bool isInternalImpl = false) : this()
{
    var __ptr = (Internal*)native.ToPointer();
    sbyte[] __value = null;
    if (__ptr->fromOnlineId != null)
    {
        __value = new sbyte[16];
        for (int i = 0; i < 16; i++)
            __value[i] = __ptr->fromOnlineId[i];
    }
    fromOnlineId = __value;
    sbyte[] __value = null;
    if (__ptr->toOnlineId != null)
    {
        __value = new sbyte[16];
        for (int i = 0; i < 16; i++)
            __value[i] = __ptr->toOnlineId[i];
    }
    toOnlineId = __value;
    byte[] __value = null;
    if (__ptr->buffer != null)
    {
        __value = new byte[508];
        for (int i = 0; i < 508; i++)
            __value[i] = __ptr->buffer[i];
    }
    buffer = __value;
    size = __ptr->size;
}
```

**Comment by tritao** [comment]
Thanks for the report, should be straightforward to fix.

**Comment by genuinelucifer** [comment]
My solution for this :-1: 
https://gist.github.com/genuinelucifer/e64cfee951a1c36f0e81
.
Please provide a review and if it is acceptable i can open a pull request.

**Comment by genuinelucifer** [comment]
Fixed by #491 . @jamesford42 @tritao Please close the issue.
